tests/test_parsers/test_docstrings/test_restructured_text.py

```diff
@@ -0,0 +1,699 @@
+    sections, errors = parse_detailed(
+        f"""
+        Docstring with one line attribute.
+
+        :{attribute_directive_name} {SOME_EXCEPTION_NAME}: {SOME_TEXT}
+        """
+    )
+    assert len(sections) == 2
+    assert sections[1].type == Section.Type.EXCEPTIONS
+    assert_annotated_obj_equal(
+        sections[1].value[0],
+        AnnotatedObject(annotation=SOME_EXCEPTION_NAME, description=SOME_TEXT),
+    )
+
+
+# -------------------------------
+# Fixture tests
+# -------------------------------
+
+
+def test_parse_module_attributes_section__expected_attributes_section():
+    """Parse attributes section in modules."""
+    obj = get_rst_object_documentation("docstring_attributes_section")
+    assert len(obj.docstring_sections) == 2
+    attr_section = obj.docstring_sections[1]
+    assert attr_section.type == Section.Type.ATTRIBUTES
+    assert len(attr_section.value) == 5
+    expected = [
+        {"name": "A", "annotation": "int", "description": "Alpha."},
+        # type annotation takes preference over docstring
+        {"name": "B", "annotation": "str", "description": "Beta."},
+        {"name": "C", "annotation": "bool", "description": "Gamma."},
+        {"name": "D", "annotation": "", "description": "Delta."},
+        {"name": "E", "annotation": "float", "description": "Epsilon."},
+    ]
+    assert [serialize_attribute(attr) for attr in attr_section.value] == expected
+
+
+def test_parse_module_attributes_section__expected_docstring_errors():
+    """Parse attributes section in modules."""
+    obj = get_rst_object_documentation("docstring_attributes_section")
+    assert len(obj.docstring_errors) == 1
+    assert "Duplicate attribute information for 'B'" in obj.docstring_errors[0]
+
+
+def test_property_docstring():
+    """Parse a property docstring."""
+    class_ = get_rst_object_documentation("docstrings.NotDefinedYet")
+    prop = class_.attributes[0]
+    sections, errors = prop.docstring_sections, prop.docstring_errors
+    assert len(sections) == 2
+    assert not errors
+
+
+def assert_parameter_equal(actual: Parameter, expected: Parameter) -> None:
+    assert actual.name == expected.name
+    assert_annotated_obj_equal(actual, expected)
+    assert actual.kind == expected.kind
+    assert actual.default == expected.default
+
+
+def assert_attribute_equal(actual: Attribute, expected: Attribute) -> None:
+    assert actual.name == expected.name
+    assert_annotated_obj_equal(actual, expected)
+
+
+def assert_annotated_obj_equal(actual: AnnotatedObject, expected: AnnotatedObject) -> None:
+    assert actual.annotation == expected.annotation
+    assert actual.description == expected.description
+
+
+def get_rst_object_documentation(dotted_fixture_subpath) -> Object:
+    return Loader(docstring_style="restructured-text").get_object_documentation(
+        f"tests.fixtures.parsing.restructured_text.{dotted_fixture_subpath}"
+    )
```

Comment:
Sure. I generally do it with the helper functions at the end, but it doesn't matter much. Another option would be to implement `__eq__()` on the section detail objects, This would allow for `assert sections[1].value == [EXPECTED_PARAM]`. I opted not to do that to keep my PR more self-contained without making changes to the other structures.